spec/02-integration/05-proxy/01-router_spec.lua

```diff
@@ -149,6 +149,58 @@ describe("Router", function()
     end)
   end)
 
+  describe("URI arguments (querystring)", function()
+
+    setup(function()
+      insert_apis {
+        {
+          name = "api-1",
+          upstream_url = "http://httpbin.org",
+          hosts = { "example.com" },
+        },
+      }
+
+      assert(helpers.start_kong())
+    end)
+
+    teardown(function()
+      helpers.stop_kong()
+    end)
+
+    it("preserves URI arguments", function()
+      local res = assert(client:send {
+        method = "GET",
+        path   = "/get",
+        query  = {
+          foo   = "bar",
+          hello = "world",
+        },
+        headers = {
+          ["Host"] = "example.com",
+        },
+      })
+
+      local body = assert.res_status(200, res)
+      local json = cjson.decode(body)
+      assert.equal("bar", json.args.foo)
+      assert.equal("world", json.args.hello)
+    end)
+
+    it("does not proxy '$is_args' if URI does not contain arguments", function()
```

Comment:
Hmm yeah I did think about it too, and wanted to explain why not, but I forgot.

In my thinking, as much as we want transparent proxying to happen (and you know how much I pushed for this), I just cannot, for the life of me, think of a use case for why someone would want to send a request with an empty querystring `/foo?`. Hence, I did not take care of this and just thought the Nginx behavior was sufficient enough in this case. If we ever receive such a requirement from a use-case, beer is on me!